example/convolution1D/src/convolution1D.cpp

```diff
@@ -0,0 +1,204 @@
+    QueueAcc queue(devAcc);
+
+    // Allocate memory host input
+    auto hostInputMemory = alpaka::allocBuf<DataType, Idx>(devHost, InputSize);
+    DataType* nativeHostInputMemory = alpaka::getPtrNative(hostInputMemory);
+
+    // Fill array with data
+    for(size_t i = 0; i < InputSize; i++)
+        nativeHostInputMemory[i] = static_cast<DataType>(i + 1);
+
+    // Allocate memory host filter
+    auto hostFilterMemory = alpaka::allocBuf<DataType, Idx>(devHost, FilterSize);
+    DataType* nativeHostFilterMemory = alpaka::getPtrNative(hostFilterMemory);
+
+    // Fill array with any data
+    for(size_t i = 0; i < FilterSize; i++)
+        nativeHostFilterMemory[i] = static_cast<DataType>(i + 1) / 10.0f;
+
+    // Allocate memory in device
+    alpaka::Buf<DevAcc, DataType, Dim, Idx> inputDeviceMemory = alpaka::allocBuf<DataType, Idx>(devAcc, InputSize);
+    alpaka::Buf<DevAcc, DataType, Dim, Idx> filterDeviceMemory = alpaka::allocBuf<DataType, Idx>(devAcc, FilterSize);
+    alpaka::Buf<DevAcc, DataType, Dim, Idx> outputDeviceMemory
+        = alpaka::allocBuf<DataType, Idx>(devAcc, static_cast<Idx>(InputSize));
+
+    // Copy input and filter (convolution kernel array) from host to device
+    alpaka::memcpy(queue, inputDeviceMemory, hostInputMemory, InputSize);
+    alpaka::memcpy(queue, filterDeviceMemory, hostFilterMemory, FilterSize);
+
+    DataType* nativeFilterDeviceMemory = alpaka::getPtrNative(filterDeviceMemory);
+    DataType* nativeInputDeviceMemory = alpaka::getPtrNative(inputDeviceMemory);
+    DataType* nativeOutputDeviceMemory = alpaka::getPtrNative(outputDeviceMemory);
+
+    size_t gridSize = InputSize;
+
+    using Vec = alpaka::Vec<Dim, Idx>;
+    using WorkDiv = alpaka::WorkDivMembers<Dim, Idx>;
+
+    auto const elementsPerThread = Vec::all(static_cast<Idx>(1));
+    auto const threadsPerGrid = gridSize;
+    WorkDiv const workDiv = alpaka::getValidWorkDiv<DevAcc>(
+        devAcc,
+        threadsPerGrid,
+        elementsPerThread,
+        false,
+        alpaka::GridBlockExtentSubDivRestrictions::Unrestricted);
+
+    // Instantiate the kernel (gpu code) function-object
+    ConvolutionKernel convolutionKernel;
+
+    // Run the kernel
+    alpaka::exec<DevAcc>(
+        queue,
+        workDiv,
+        convolutionKernel,
+        nativeInputDeviceMemory,
+        nativeFilterDeviceMemory,
+        nativeOutputDeviceMemory,
+        InputSize,
+        FilterSize);
+    alpaka::wait(queue);
+
+    // Allocate memory on host
+    alpaka::Buf<alpaka::DevCpu, DataType, Dim, Idx> resultGpuHost
+        = alpaka::allocBuf<DataType, Idx>(devHost, InputSize);
+    // Copy from device memory to host
+    alpaka::memcpy(queue, resultGpuHost, outputDeviceMemory, InputSize);
+
+    bool allEqual{true};
+    // Print result array at the host
+    for(size_t i{0}; i < InputSize; i++)
+    {
+        std::cout << "output[" << i << "]:" << std::setprecision(3) << resultGpuHost[i] << std::endl;
```

Comment:
No need for a stream flush in every iteration.
```suggestion
        std::cout << "output[" << i << "]:" << std::setprecision(3) << resultGpuHost[i] << "\n";
```

Reply:
Ok, for preventing flushing.